pkg/deploy/strategy/support/lifecycle.go

```diff
@@ -374,13 +374,14 @@ func makeHookPod(hook *deployapi.LifecycleHook, deployment *kapi.ReplicationCont
 		Spec: kapi.PodSpec{
 			Containers: []kapi.Container{
 				{
-					Name:         HookContainerName,
-					Image:        baseContainer.Image,
-					Command:      exec.Command,
-					WorkingDir:   baseContainer.WorkingDir,
-					Env:          mergedEnv,
-					Resources:    resources,
-					VolumeMounts: volumeMounts,
+					Name:            HookContainerName,
+					Image:           baseContainer.Image,
+					ImagePullPolicy: baseContainer.ImagePullPolicy,
+					Command:         exec.Command,
+					WorkingDir:      baseContainer.WorkingDir,
+					Env:             mergedEnv,
+					Resources:       resources,
+					VolumeMounts:    volumeMounts,
```

Comment:
@VanRoy created https://github.com/openshift/origin/issues/12103 to make sure we follow up on it.  No need to add it here.